client/launchpad/hooks/use-task-completed-notice.ts

```diff
@@ -0,0 +1,55 @@
+import page from '@automattic/calypso-router';
+import { updateLaunchpadSettings, useLaunchpad } from '@automattic/data-stores';
+import { useI18n } from '@wordpress/react-i18n';
+import { useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'calypso/state';
+import { successNotice } from 'calypso/state/notices/actions';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+
+interface TaskCompleteNoticeOptions {
+	taskSlug: string;
+	enabled?: boolean;
+	noticeId: string;
+	noticeText: string;
+	noticeDuration?: number;
+}
+
+export const useTaskCompletedNotice = ( {
```

Comment:
This is a nice wee hook, I like this approach 👍 

Since the completion logic is unique for each task, I sort of wanted to have a different hook name, that made it clear that this hook was appropriate when the task is completed on navigation.

```suggestion
export const useTaskCompletedOnNavigationNotice = ( {
```

But this seems too clunky. So I think just leave it for now until something better becomes obvious.